### PR TITLE
To do list structure

### PR DESCRIPTION
**Pull request by Nahid1911**
### _*Requirements:*_

1. install webpack,
2. used HTML template.
3. Created array with objectives.
4. Write a function to iterate over the tasks array and populate an HTML list item element for each task.
5. On page load render the dynamically created list of tasks in the dedicated placeholder. The list should appear in order of the index values for each task.
6. Create a style.css and set rules for the To Do List. CSS must be loaded by Webpack Style/CSS Loader. Your list should be a clone of the part of the minimalist project.
7. All your source files (index.html, index.js and style.css) must be located in /src directory and your distribution files will be generated by webpack and served by webpack dev server from /dist folder.

**for more details about requirements please click [HERE](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m1_list_structure.md)**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,14 +1,51 @@
...
+      <input for ="job${index}" id="inputField" type="checkbox" class ="checkBox">
+      <p id ="job${index}" class ="pInsideLi">${task.description}</p>
+      </label>
+      <button  class="btnAll threeDots">
+       <i class="fa fa-ellipsis-h"></i>
+      </button>
+      </li>
+    `;
+  return content;
+}
```
Comment (abdulhamiid) [inline src/index.js]: Good job with additional functionality 💯 

However, as this is a basic project, we are still required to fulfill all the project's core requirements. 

According to the [project requirement](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m1_list_structure.md), please kindly implement the following 🤗👍

- Create an **index.js** file and set an array of some simple *to do tasks* (array of objects). Each task object should contain three keys: 
  1. description [`string`].
  2. completed [`bool`].
  3. index: [`number`].

- On page load render the dynamically created list of tasks in the dedicated placeholder. The list should appear in order of the `index` values for each task.

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi @Nahid1911 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! :running_man: :runner: 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are very descriptive
- ✔️ Linters checks are passing
- ✔️ Use GitHub Flow
- ✔️ Meaningful commit messages.

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (Nahid1911) [reply src/index.js]: @abdulhamiid 
resolve the requirement.

File: src/index.js [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
+</li>
+<li class ="commonClass">
+<label class ="checkBoxLabel">
+<input for ="job2" id="inputField" type="checkbox" class ="checkBox">
+<p id ="job2" class ="pInsideLi">${tasks[1].description}</p>
+</label>
+<button  class="btnAll threeDots">
+ <i class="fa fa-ellipsis-h"></i>
+</button>
+</li>
```
Comment (OyePriscilla) [inline src/index.js]: - [ ] Kindly manipulate the DOM dynamically as raised earlier, you only need the first portion of the code written above to achieve this. More so, there is a need to loop through the task array in order to access the content of the array. 
```suggestion
const target = document.getElementById('listContainer');
for (let i = 0; i < tasks.length; i += 1) {
  target.innerHTML += `
<li class ="commonClass">
<label class ="checkBoxLabel">
<input for ="job" id="inputField" type="checkbox" class ="checkBox">
<p id ="job" class ="pInsideLi">${tasks[i].description}</p>
</label>
<button  class="btnAll threeDots">
 <i class="fa fa-ellipsis-h"></i>
</button>
</li>
`;
}
```

- [ ] Kindly ensure that the list appear in order of the index values for each task.

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Nahid1911 👏 👏, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

so-close-so-close-seth-meyers

 ## Highlights
Use webpack to bundle JavaScript ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [optional] Kindly remove the empty live demo header in your README file if you don't have a live demo link at the moment. 👍 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 2 above)
Comment (Nahid1911) [reply src/index.js]: resolved

**Review by ABDULALI3468 — APPROVED** [review]
Hi @Nahid1911 , 👍 

# APPROVED 💯 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
☑️Implemented changes asked by the previous CR. 
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification.

> * This is My **GitHub**: GitHub followers
> * This is My **Linkedin**: [Linkedin Badge](https://www.linkedin.com/in/abdul-ali-5400bb216/)&nbsp;
> * This is My **Twitter**: [Twitter Badge](https://twitter.com/mrabdul_ali)&nbsp;

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**